src/aggregated_queries.rs

```diff
@@ -33,7 +36,20 @@ impl AggregatedQueriesPerLanguageBuilder {
         let query = Arc::new(Query::new(language.language(), &query_text).unwrap());
         assert!(query.pattern_count() == pattern_index_lookup.len());
         AggregatedQueriesPerLanguage {
-            pattern_index_lookup,
+            pattern_index_lookup: pattern_index_lookup
+                .into_iter()
+                .map(
+                    |(rule_index, rule_listener_index, capture_name_if_per_capture)| {
+                        (
+                            rule_index,
+                            rule_listener_index,
+                            capture_name_if_per_capture.map(|capture_name| {
+                                query.capture_index_for_name(&capture_name).unwrap()
```

Comment:
The bug was that I was resolving capture index against the _individual_ parser listener queries but clearly that is not always the same as the capture index for that same capture name in the _aggregated_ query (since I was sometimes only seeing violations from a subset of the rules printed), so basically defer resolving capture index -> capture name until we have the aggregated query to "interrogate"